api/resource.rb

```diff
@@ -26,52 +26,94 @@ class Resource < Api::Object::Named
     module Properties
       include Api::Object::Named::Properties
 
+      # [Required] A description of the resource that's surfaced in provider
+      # documentation.
       attr_reader :description
-      # GCP kind, e.g. `compute#disk`
-      attr_reader :kind
-      # URI: relative to `@api.base_url` or absolute
+      # [Required] (Api::Resource::ReferenceLinks) Reference links provided in
+      # downstream documentation.
+      attr_reader :references
+      # [Required] The GCP "relative URI" of a resource, relative to the product
+      # base URL. It can often be inferred from the `create` path.
       attr_reader :base_url
-      # URL to use for creating the resource. If not specified, the
-      # collection url (when create_verb is default or :POST) or
-      # self_link (when create_verb is :PUT) is used instead.
+
+      # ====================
+      # Common Configuration
+      # ====================
+      #
+      # [Optional] The minimum API version this resource is in. Defaults to ga.
+      attr_reader :min_version
+      # [Optional] If set to true, don't generate the resource.
+      attr_reader :exclude
+      # [Optional] If set to true, the resource is not able to be updated.
+      attr_reader :input
+      # [Optional] If set to true, this resource uses an update mask to perform
+      # updates. This is typical of newer GCP APIs.
+      attr_reader :update_mask
+      # [Optional] If set to true, the object has a `self_link` field. This is
+      # typical of older GCP APIs.
+      attr_reader :has_self_link
+
+      # ====================
+      # URL / HTTP Configuration
+      # ====================
+      #
+      # [Optional] The "identity" URL of the resource.. Defaults to:
+      # * collection url when the create_verb is :POST
```

Comment:
This is a bit confusing since collection url hasn't been defined yet- I think base url accomplishes the same thing